Source/Core/Core/WC24PatchEngine.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+// WC24PatchEngine
+// Allows for replacing URLs used in WC24 requests
+
+#include <algorithm>
+#include <array>
+#include <fmt/format.h>
+
+#include "Common/StringUtil.h"
+
+#include "Core/CheatCodes.h"
+#include "Core/CommonTitles.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/ConfigManager.h"
+#include "Core/WC24PatchEngine.h"
+
+namespace WC24PatchEngine
+{
+static std::array<u64, 12> s_wc24_channels{
+    // Nintendo Channel
+    Titles::NINTENDO_CHANNEL_NTSC_U,
+    Titles::NINTENDO_CHANNEL_NTSC_J,
+    Titles::NINTENDO_CHANNEL_PAL,
+    Titles::FORECAST_CHANNEL_NTSC_U,
+    Titles::FORECAST_CHANNEL_NTSC_J,
+    Titles::FORECAST_CHANNEL_PAL,
+    Titles::NEWS_CHANNEL_NTSC_U,
+    Titles::NEWS_CHANNEL_NTSC_J,
+    Titles::NEWS_CHANNEL_PAL,
+    Titles::EVERYBODY_VOTES_CHANNEL_NTSC_U,
+    Titles::EVERYBODY_VOTES_CHANNEL_NTSC_J,
+    Titles::EVERYBODY_VOTES_CHANNEL_PAL,
+};
+
+static std::vector<NetworkPatch> s_patches;
+
+bool DeserializeLine(const std::string& line, NetworkPatch* patch)
+{
+  const std::vector<std::string> items = SplitString(line, ':');
+
+  if (items.size() != 3)
+    return false;
+
+  patch->source = items[0];
+  patch->replacement = items[1];
+
+  if (!TryParse(items[2], &patch->is_kd))
+    return false;
+
+  return patch;
+}
+
+void LoadPatchSection(const Common::IniFile& ini)
+{
+  std::vector<std::string> lines;
+  NetworkPatch patch;
+  ini.GetLines("WC24Patch", &lines);
+
+  for (std::string& line : lines)
+  {
+    if (line.empty())
+      continue;
+
+    if (line[0] == '$')
+    {
+      patch.name = line.substr(1, line.size() - 1);
+    }
+    else
+    {
+      if (DeserializeLine(line, &patch))
+      {
+        s_patches.push_back(patch);
+      }
+    }
+  }
+
+  ReadEnabledAndDisabled(ini, "WC24Patch", &s_patches);
+}
+
+void LoadPatches()
+{
+  const auto& sconfig = SConfig::GetInstance();
+  // We can only load WC24 Channels.
+  if (!IsWC24Channel())
+    return;
+
+  Common::IniFile ini;
+  // If WiiLink is enabled then we load the Default Ini as that has the WiiLink URLs.
+  if (Config::Get(Config::MAIN_WII_WIILINK_ENABLE))
+    ini = sconfig.LoadDefaultGameIni();
+  else
+    ini = sconfig.LoadLocalGameIni();
```

Comment:
This logic here also feels a bit jank now that I think about it... I would expect user patches to apply in addition to the system ones, regardless of if the WiiLink option is enabled or not. Though I'd be okay with that being a follow-up by either you or someone else, I'm not expecting every user to suddenly start writing URL replacements into their INIs.

Reply:
The logic I was trying to get at here is that the WiiLink URL's will be contained within the INI's we pack. If Enable WiiLink is off, it would allow for users to put in their own replacement URL if they please. I just didn't want any conflicts between the INI's URL's